modules/navier_stokes/src/variables/INSFVVelocityVariable.C

```diff
@@ -65,6 +65,41 @@ INSFVVelocityVariable::INSFVVelocityVariable(const InputParameters & params) : I
 }
 
 #ifdef MOOSE_GLOBAL_AD_INDEXING
+
+const ADReal &
+INSFVVelocityVariable::getExtrapolatedBoundaryFaceValue(const FaceInfo & fi) const
+{
+  if (_two_term_boundary_expansion)
+    if (isFullyDevelopedFlowFace(fi))
+    {
+      const auto & tup = Moose::FV::determineElemOneAndTwo(fi, *this);
+      const Elem * const elem = std::get<0>(tup);
+      const Point vector_to_face = std::get<2>(tup) ? (fi.faceCentroid() - fi.elemCentroid())
+                                                    : (fi.faceCentroid() - fi.neighborCentroid());
+
+      _temp_face_value = uncorrectedAdGradSln(fi) * vector_to_face + getElemValue(elem);
+
+      return _temp_face_value;
+    }
+
+  return INSFVVariable::getExtrapolatedBoundaryFaceValue(fi);
```

Comment:
Let's then also put this in an `else` branch. I think it's more readable that way. It's nice to reduce indentation, but sometimes that can make the code less clear